core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -301,6 +304,9 @@ public boolean deleteBlock(long blockId) throws IOException {
       }
       deleteBlockId(blockId);
     } else {
+      if (mRemovedBlockIdList.contains(blockId)) {
```

Comment:
Hmm I don't actually think it should happen, since it's only if you've locked a block after deleting it, which wouldn't happen. So I'll remove it.